src/gwt/src/org/rstudio/studio/client/workbench/views/presentation/PresentationPane.java

```diff
@@ -295,7 +295,12 @@ public void hideBusy()
       refreshButton_.setVisible(true);
       progressButton_.setVisible(false);
    }
-   
+
+   @Override
+   public void setFocus()
+   {
+      slideNavigationMenu_.focus();
```

Comment:
There's a repeated pattern here where we set focus to what (presumably) is the first focusable command/button. I worry a little that this is going to be tricky to maintain, because if you permute the order of the commands you're going to wind up making F6 sending command to some random place in the middle, and you won't notice unless you happen to test F6. I wonder if we should have a method which performs a heuristic instead to find the first focusable element? (Gary might also have some thoughts on the best rules by which to select an element for focus)

Reply:
I talked to @gtritchie briefly about this before starting and he suggested determining the best place per pane to select focus and setting it accordingly. I did run into some issues with determining which element this is in every case, and more often than not defaulted to the first focusable element.

It might be tricky to write a generic approach since we don't typically save the button as a part of the pane's class and there is no rule that a pane needs to have a focusable button.

Reply:
I haven't looked at this PR yet (want to build it and play with it).

That said, there is a technique I added in the modal dialog code that dynamically determines the first (and last) focusable element contained with a parent element, so we might be able to leverage something similar here and set a reasonable default.

https://github.com/rstudio/rstudio/blob/7fbec0b01eaccab6970d2c295f73e086ba2446eb/src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java#L751-L773

Probably do still want a way for each pane to override that; for the dialogs, that is done via the `focusInitialControl()` method. By default it just sets focus to the programmatically-determined first control, but some dialogs change this.